azure/scope/machinepool.go

```diff
@@ -51,7 +53,10 @@ import (
 // ScalesetsServiceName is the name of the scalesets service.
 // TODO: move this to scalesets.go once we remove the usage in this package,
 // added here to avoid a circular dependency.
-const ScalesetsServiceName = "scalesets"
+const (
+	ScalesetsServiceName = "scalesets"
+	CustomDataHash       = "CustomDataHash"
```

Comment:
let's move this to https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/const.go where the other annotations are and use the same format and prefix, with a code comment that explains how it's used

```
sigs.k8s.io/cluster-api-provider-azure-vmss-custom-data-hash
```